### PR TITLE
Join UseSiteListAsLocation features in trustSitelists

### PR DESCRIPTION
**Pull request by amaltaro**
Attempt to merge these 2 flags into TrustSitelists only, that means the special handling of workqueue elements was transferred from UseSiteListAsLocation to TrustSitelists.

At the job level, TrustSitelists will force the (siteWhitelist - siteBlacklist) as DESIRED_Sites.

Still needs to be tested.

### CODE REVIEW

File: src/python/WMCore/WorkQueue/DataStructs/WorkQueueElement.py [hunk 1]
```diff
@@ -213,7 +213,7 @@ def statusMetrics(self):
     def passesSiteRestriction(self, site):
         """Takes account of white & black list, and input data to work out
         if site can run the work"""
-        # Don't check anything if useSiteListAsLocation/trusSiteLists is enabled
+        # Don't check anything if trustSitelists is enabled
         if self['NoLocationUpdate']:
```
Comment (hufnagel) [inline src/python/WMCore/WorkQueue/DataStructs/WorkQueueElement.py]: How is this NoLocationUpdate set ? It defaults to False, while standard TrustSiteLists modus would need it at True, ie. we really don't care about file locations at all.

Comment (amaltaro) [inline src/python/WMCore/WorkQueue/DataStructs/WorkQueueElement.py]: It's set in the WorkQueue start policies, e.g.:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WorkQueue/Policy/Start/Block.py#L57

though I just noticed I was mixing an update there as well. Updating the code

**Comment by amaltaro** [comment]
Fixed the conflicts, however I have no idea how to squash these 2 commits now, since there is a third commit  (merge PR 6557) in between
```
7df150666b88caed428fda9f18fe0050855310e4 (HEAD, origin/trustSite-patch2, trustSite-patch2) Merging from master
dc3293521b393b1661e0e9decb7919a11b42b5c1 (upstream/master, origin/master, origin/HEAD, master) Merge pull request #6557 from amaltaro/phedexGroup-fix
2b87f1d18a85abddadbaa92b55d8e757ecddd292 Join UseSiteListAsLocation features in trustSitelists
```

any ideas? :-) Though it shouldn't harm just leaving it as it is.

**Comment by ericvaandering** [comment]
You can also re-order commits with git rebase -i

**Comment by amaltaro** [comment]
Overseeded by #6569

**Pull request closed without merging**